src/talos/RunHailFiltering.py

```diff
@@ -635,40 +621,103 @@ def filter_by_consequence(mt: hl.MatrixTable) -> hl.MatrixTable:
+        | ((de_novo_matrix.GT.is_hom_var()) & (de_novo_matrix.PL[0] < min_gq))
+        | ((de_novo_matrix.GT.is_het()) & (de_novo_matrix.PL[0] < min_gq))
+        # single added condition here
+        | ((de_novo_matrix.GT.is_het()) & (de_novo_matrix.AD[1] < (0.20 * de_novo_matrix.DP))),
+        keep=False,
+    )
 
-    de_novo_matrix = de_novo_matrix.annotate_entries(
-        PL=hl.case()
-        .when(~hl.is_missing(de_novo_matrix.PL), de_novo_matrix.PL)
-        .when((de_novo_matrix.GT.is_non_ref()) | (hl.is_missing(de_novo_matrix.GQ)), hl.missing('array<int32>'))
-        .default([0, de_novo_matrix.GQ, 1000]),
+    # create a trio matrix (variant rows, trio columns)
+    tm = hl.trio_matrix(de_novo_matrix, pedigree, complete_trios=True)
+
+    kid = tm.proband_entry
+    dad = tm.father_entry
+    mom = tm.mother_entry
+
+    # Updated for hemizygous child calls to not require a call in uninvolved parent
+    has_candidate_gt_configuration = (
+        (tm.locus.in_autosome_or_par() & kid.GT.is_het() & dad.GT.is_hom_ref() & mom.GT.is_hom_ref())
+        | (tm.locus.in_x_nonpar() & ~tm.is_female & ((kid.GT.is_hom_var()) | (kid.GT.is_het())) & mom.GT.is_hom_ref())
+        | (tm.locus.in_x_nonpar() & tm.is_female & kid.GT.is_het() & mom.GT.is_hom_ref() & dad.GT.is_hom_ref())
+        | (tm.locus.in_y_nonpar() & ((kid.GT.is_hom_var()) | (kid.GT.is_het())) & dad.GT.is_hom_ref())
+        | (tm.locus.in_mito() & kid.GT.is_hom_var() & mom.GT.is_hom_ref())
+    )
+
+    # require AD & PL for called variants, just not always for HomRef
+    kid_ad_ratio = kid.AD[1] / hl.sum(kid.AD)
+
+    # Try to get these all to an expected value of 0.5
+    dp_ratio = (
+        hl.case()
+        .when((tm.locus.in_x_nonpar()) & (~tm.is_female), kid.DP / mom.DP)  # Because mom is diploid but kid is not
+        .when((tm.locus.in_y_nonpar()) & (~tm.is_female), kid.DP / (2 * dad.DP))
+        .when(tm.locus.in_mito(), kid.DP / (2 * mom.DP))
+        .when(tm.locus.in_x_nonpar() & tm.is_female, (kid.DP / (mom.DP + dad.DP)) * (3 / 4))  # Because of hemi dad
+        .default(kid.DP / (mom.DP + dad.DP))
     )
 
-    dn_table = hl.de_novo(
-        de_novo_matrix,
-        pedigree,
-        pop_frequency_prior=de_novo_matrix.info.gnomad_af,
-        ignore_in_sample_allele_frequency=True,
-        max_parent_ab=config_retrieve(['RunHailFiltering', 'max_parent_ab'], 0.05),
+    # horribly simplified - we don't have the PL or AD for any WTs, so we're really fudging the main parts
+    # I've also dropped the requirements for different confidence levels, we're treating Low/Medium/High equally
+    tm = tm.annotate_entries(
+        de_novo_tested=hl.case()
+        .when(~has_candidate_gt_configuration, MISSING_INT)
+        .when(min_alt_depth > kid.AD[1], MISSING_INT)
+        .when(min_gq > kid.GQ, MISSING_INT)
+        .when((dp_ratio < min_dp_ratio) | (kid_ad_ratio < min_child_ab), MISSING_INT)
+        .when(kid_ad_ratio > ratio_0_2, ONE_INT)
```

Comment:
This seems a bit redundant given the OR above? but it does not seem like it should be hard coded in any case?
```suggestion
        .when(kid_ab > min_child_ab, ONE_INT)
```

Reply:
Yeah, happy to drop this. This comes from the original implementation which applied a confidence tier depending on which thresholds were satisfied. I'm dropping all the confidence tiers in favour of YES/NO for our needs, which means we can probably thin out some of the conditional tests

Reply:
I've addressed this by inverting the above condition - if the `dp_ratio` is higher than the `min_dp_ratio`, or `kid_ab` is higher than `min_child_ab`, the ✅ is applied. This redundant second check is removed